libcxx/test/std/language.support/support.dynamic/new.delete/new.delete.placement/new.pass.cpp

```diff
@@ -15,19 +15,30 @@
 
 int A_constructed = 0;
 
-struct A
-{
-    A() {++A_constructed;}
-    ~A() {--A_constructed;}
+struct A {
+  A() { ++A_constructed; }
+  ~A() { --A_constructed; }
 };
 
-int main(int, char**)
-{
-    char buf[sizeof(A)];
+TEST_CONSTEXPR_OPERATOR_NEW void test_direct_call() {
+  assert(::operator new(sizeof(int), &A_constructed) == &A_constructed);
```

Comment:
```suggestion
  assert(::operator new(sizeof(A), &A_constructed) == &A_constructed);
```

Isn't that less confusing? Any reason you used `sizeof(int)`?

Reply:
> Any reason you used `sizeof(int)`?

The reason is that this line (ditto in `new_array.pass.cpp`) tested taking address of the global variable (of static storage duration) `A_constructed` whose type is `int`, not any `A` object.

Reply:
Ack, I read too fast. Thanks!